sdk/src/asset_handlers/bmff_io.rs

```diff
@@ -251,7 +257,7 @@ pub(crate) struct BoxInfoLite {
     pub size: u64,
 }
 
-fn read_box_header_ext(reader: &mut dyn CAIRead) -> Result<(u8, u32)> {
+fn read_box_header_ext<R: Read + Seek + ?Sized>(reader: &mut R) -> Result<(u8, u32)> {
```

Comment:
if only we could do this everywhere!

Reply:
It may be worth trying again.  I think the issue I had was was with Sync and Send.